analyses/chromosomal-instability/02a-plot-chr-instability-heatmaps.Rmd

```diff
@@ -56,9 +56,12 @@ Import color palettes.
 ```{r}
 # Import standard color palettes for project
 histology_label_mapping <- readr::read_tsv(
-  file.path(figures_dir, "palettes", "histology_label_color_table.tsv")
+  file.path(figures_dir, "palettes", "histology_label_color_table.tsv"), 
+  # Read in each column as a specific type - skip extra columns
+  col_types = paste0(c("c", rep("-", 6), "f", "i", "c"), collapse = "")
```

Comment:
Do you need this? I guess it is helpful for skipping columns, but the factor will be automatically converted when you call fct_reorder, and the order should get read in correctly as numeric. You are implicitly specifying column order, which is something I try to avoid as it can be brittle.

Reply:
I agree that I'm not happy about this thing which is rigged with respect to column order. However, fct_reorder will not as the table here is read in. `display_group` does not get read in as a factor which is what is causing the problem. 

I can instead get rid of this `col_types` line and add back a `mutate` where we force it to be a factor first and then also add back a `select` to get to the four columns we want. (this is what I had previously. It includes more steps but it is less brittle with respect to column order).

Reply:
Hmm, I've never had `forcats` complain about being given a character vector. Maybe I haven't used whatever old version we currently have.

Reply:
I switched back to using a `select` and `as.factor()`. It may be more official "step" but it is more robust.

Reply:
Okay. I must be imaging things. I dropped the as.factor and it works fine. Never mind everything I said. will push that change.